azure/functions/decorators/blob.py

```diff
@@ -17,7 +17,10 @@ def __init__(self,
                  **kwargs):
         self.path = path
         self.connection = connection
-        self.source = source.value if source else None
+        if type(source) is BlobSource:
+            self.source = source.value if source else None

```

Comment:
If we are checking the blobsource, it should always have a value right? Do we need another condition here?